LoopKitUI/Views/CardList/TimePicker.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  TimePicker.swift
+//  LoopKitUI
+//
+//  Created by Michael Pangburn on 4/23/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+
+/// Binds a `TimeInterval` describing an offset from midnight to a selected picker value.
+///
+/// For example, a value of `0` corresponds to midnight, while a value of `7200` corresponds to 2:00 AM.
+///
+/// The offset is not relative to the current date. Use `TimePicker` for selecting the time of a recurring daily event, _not_ for selecting the time at which an alarm should sound later this afternoon.
+public struct TimePicker: View {
+    @Binding private var offsetFromMidnight: TimeInterval
+    private let allValues: [TimeInterval]
+    private let fixedMidnight = Calendar.current.startOfDay(for: Date(timeIntervalSinceReferenceDate: 0))
```

Comment:
This assumes the user isn't in a timezone that observes any clock shifts on 1 January 2001.  It's a safe assumption probably?  Previously existing code made sure to do this kind of math using a fixed offset timezone, which also works, but does require that the dateformatter use a timezone other than the user's current one, and assumes that formatting will still work properly. I'm not sure which assumption is better, so I'm ok with this.

Reply:
Heh, yeah, I also wrote this with the thought of "it's a safe assumption probably"—I believe it's minimally true for [DST-related clock shifts](https://en.wikipedia.org/wiki/Daylight_saving_time_by_country).